compiler/AST/AggregateType.cpp

```diff
@@ -2159,12 +2159,21 @@ bool AggregateType::needsConstructor() const {
 
   AggregateType* thisNC = const_cast<AggregateType*>(this);
 
-  if (initializerStyle == DEFINES_CONSTRUCTOR)
+  if (initializerStyle == DEFINES_CONSTRUCTOR) {
     return true;
-  else if (fUserDefaultInitializers && this != dtObject)
+  } else if (fUserDefaultInitializers == false) {
+    // Don't allow constructors for internal/standard types, except for object
+    ModuleSymbol* mod = thisNC->getModule();
+    if (this == dtObject) {
+      return true;
+    } else if (mod->modTag == MOD_INTERNAL || mod->modTag == MOD_STANDARD) {
+      return false;
+    }
+  } else if (fUserDefaultInitializers && this != dtObject) {
```

Comment:
The `fUserDefaultInitializers` check is now unnecessary for this branch, since the new branch eliminates it being set to false.  You could add an assert for it in the branch if you were worried about it